source/classes.tex

```diff
@@ -36,7 +36,7 @@
 
 \begin{bnf}
 \nontermdef{class-head}\br
-    class-key \opt{attribute-specifier-seq} class-head-name \opt{class-virt-specifier} \opt{base-clause}\br
+    class-key \opt{attribute-specifier-seq} class-head-name \opt{class-property-specifier-seq} \opt{base-clause}\br
```

Comment:
It looks like a cleaner fix would be to move the optional _class-property-specifier_ onto the _class-head-name_ production, but that is probably a bigger change than should be considered editorial.

Reply:
Why would that be a "cleaner fix"?  What's the "fix" part here?  Why cleaner?

Reply:
We have two productions, but the optional class-property-specifier-seq is valid only after a class-head-name.  We could move that optional element into the class-head-name and then the optionality of the class-head-name itself becomes more apparent.

If we move the optional class-property-specifier-seq into the class-head-name, we could even simplify the grammar further by making the class-head-name itself optional, needing only one grammar production and not two to say the same thing.

I have confirmed (to the best of my ability) that there are no other uses of the class-head-name production that would become confused by having this extra optional component suddenly showing up.

Reply:
If we ever put more things into the class-property-specifier-seq, in particular things that need name lookup (e.g. when the "bool parameter" stuff shows up in another context), then [basic.scope.namespace] p1 and [basic.scope.class] p1 would go wrong. Also [expr.prim.id.qual] p2.1 .

Looks like such a grammar refactoring would not be forward-evolution-friendly.

To sum up, I think there is nothing to act upon in this editorial pull request.